File: code_combat/8_Cloudrip_Mountain/392-The_Two_Flowers/the_two_flowers.py
def summonSoldiers():
    if hero.gold >= hero.costOf("soldier"):
        hero.summon("soldier")


def commandSoldiers():
    army = hero.findByType("soldier")
    for soldier in army:
        enemy = soldier.findNearestEnemy()
        if enemy and soldier.distanceTo(enemy) < 10:
            hero.command(soldier, "attack", enemy)
        else:
            hero.command(soldier, "defend", peasant)


def pickUpNearestCoin():
    coin = hero.findNearestItem()
    if coin:
        hero.move(coin.pos)


peasant = hero.findByType("peasant")[0]

while True:
    summonSoldiers()
    commandSoldiers()
    pickUpNearestCoin()

File: projects/code_combat/8_Cloudrip_Mountain/392-The_Two_Flowers/theTwoFlowers.js
// If the peasant is damaged, the flowers will shrink!

function summonSoldiers() {
    if (hero.gold >= hero.costOf("soldier")) {
        hero.summon("soldier");
    }
}

// Define the function: commandSoldiers
function commandSoldiers() {
    var army = hero.findByType("soldier");
    for (var i = 0; i < army.length; i++) {
        var soldier = army[i];
        var enemy = soldier.findNearestEnemy();
        if (enemy && soldier.distanceTo(enemy) < 10) {
            hero.command(soldier, "attack", enemy);
        } else {
            hero.command(soldier, "defend", peasant);
        }
    }
}

// Define the function: pickUpNearestCoin
function pickUpNearestCoin() {
    var coin = hero.findNearestItem();
    if (coin)
        hero.move(coin.pos);
}

var peasant = hero.findByType("peasant")[0];

while(true) {
    summonSoldiers();
    commandSoldiers();
    pickUpNearestCoin();
    
}
